src/proposer/proposer_rpc.cpp

```diff
@@ -220,17 +217,15 @@ class ProposerRPCImpl : public ProposerRPC {
   }
 
   UniValue propose(const JSONRPCRequest &request) const override {
-    std::shared_ptr<CWallet> const wallet = GetWalletForJSONRPCRequest(request);
-    CWallet *const pwallet = wallet.get();
+    CWallet *const pwallet = GetWalletForJSONRPCRequest(request).get();
```

Comment:
Sorry, I actually meant something different.
I asked if we need `pwallet` variable, because we already have shared pointer `wallet`.
I think it is better to keep it as shared pointer. Because once you assign it to a variable - this variable "holds" this instance and it won't be deleted until the end of the block. While in your new version it can happen. Potentially.

Although I doubt this is possible in this concrete place - I still think it is a code smell, to use raw pointer, when you already have shared pointer.

Reply:
The only point where the raw pointer is used is inside `EnsureWalletIsAvailable`, is bitcoin code, ain't changing it. The other usages are actually references so there should not be any issue with those.

Reply:
Consider this:
```cpp
{
  shared_ptr<T> p = GetP();
  do_something_with_p_ref(*p);
} // p is guaranteed to be alive upon here
```

vs this:

```cpp
{
  T* p = GetP().get();
  // p can become garbage here already
  do_something_with_p_ref(*p);
}
```
What I mean is that second variant is _potentially_ more dangerous. Because reference count can reach zero (on another thread for example) and `p` will become dangling pointer

Reply:
I understand, I'll change it even if honestly to me it seems a bit of an excessive concern.